### PR TITLE
[NEW] Adds a Keyboard Shortcut option to the flextab

### LINKED ISSUE #46: Create Help / Shortcuts modal window
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/18752135-create-help-shortcuts-modal-window?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by cnash**
@RocketChat/core 

Closes #46

I added a button to the "FlexTabBar" on the side that opens a tab for displaying the Keyboard Shortcuts. The shortcuts are defined in en.i18n.json (sorry, I only know english). They are in a list, so you can add more in the future without modifying the template.

### CODE REVIEW

**Comment by rodrigok** [comment]
Hi @cnash, can you paste screen shots of your implementation here?

**Comment by cnash** [comment]
screen shot 2017-02-03 at 4 55 12 pm

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -705,6 +705,18 @@
...
+    "Items": [
+      { "Description": "Open Channel / User search", "Keys": "<kbd>Ctrl</kbd> + <kbd>p</kbd>"},
+      { "Description": "Edit previous message", "Keys": "<kbd>Up Arrow</kbd>"},
+      { "Description": "Move to the beginning of the message", "Keys": "<kbd>Command</kbd> (or <kbd>Alt</kbd>) + <kbd>Left Arrow</kbd>"},
+      { "Description": "Move to the beginning of the message", "Keys": "<kbd>Command</kbd> (or <kbd>Alt</kbd>) + <kbd>Up Arrow</kbd>"},
+      { "Description": "Move to the end of the message", "Keys": "<kbd>Command</kbd> (or <kbd>Alt</kbd>) + <kbd>Right Arrow</kbd>"},
+      { "Description": "Move to the end of the message", "Keys": "<kbd>Command</kbd> (or <kbd>Alt</kbd>) + <kbd>Down Arrow</kbd>"},
+      { "Description": "New line in message compose input", "Keys": "<kbd>Shift</kbd> + <kbd>Enter</kbd>"}
+    ]
+  },
```
Comment (marceloschmidt) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Heads up! We never used structured json for translations. Not sure how Lingohub will behave.

Comment (cnash) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Do I need to change this then? I was hoping to use the list structure to make it easier to add more shortcuts in the future.

Comment (marceloschmidt) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: I am not sure :( let's see if someone has more info on this one regarding lingohub.

Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: I think we will need to change this, LingoHub won't parse this.

Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Let's make it a fixed list, with unique keys, rather than a loop to iterate over an array.

Comment (cnash) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Okay

**Comment by karlprieb** [comment]
@cnash Hey! Do you still have interest in finishing this PR?

**Comment by cnash** [comment]
Yeah. Sorry. Work got crazy, and then I forgot about this. I'll try to fix it up this evening.

File: packages/rocketchat-lib/client/defaultTabBars.js [hunk 2]
```diff
@@ -33,3 +33,13 @@ RocketChat.TabBar.addButton({
...
 });
+
+RocketChat.TabBar.addButton({
+	groups: ['channel', 'privategroup', 'directmessage'],
+	id: 'keyboard-shortcut-list',
+	i18nTitle: 'Keyboard_Shortcuts.Title',
+	icon: 'icon-keyboard',
+	template: 'keyboardShortcuts',
+	order: 4
+});
```
Comment (cnash) [inline packages/rocketchat-lib/client/defaultTabBars.js]: I keep adding newlines to this file, and @codacy-bot keeps complaining.

File: packages/rocketchat-lib/client/defaultTabBars.js [hunk 3]
```diff
@@ -33,3 +33,13 @@ RocketChat.TabBar.addButton({
 	template: 'uploadedFilesList',
 	order: 3
 });
+
+RocketChat.TabBar.addButton({
+	groups: ['channel', 'privategroup', 'directmessage'],
+	id: 'keyboard-shortcut-list',
+	i18nTitle: 'Keyboard_Shortcuts.Title',
+	icon: 'icon-keyboard',
```
Comment (rodrigok) [inline packages/rocketchat-lib/client/defaultTabBars.js]: @karlprieb Please fix this icon on new design

**Pull request merged**